src/handlers/token_handler.erl

```diff
@@ -0,0 +1,32 @@
+-module(token_handler).
+
+-export([init/3,
+         allowed_methods/2,
+         content_types_provided/2,
+         is_authorized/2,
+         to_json/2]).
+
+init(_Transport, _Req, []) ->
+    {upgrade, protocol, cowboy_rest}.
+
+allowed_methods(Req, State) ->
+    {[<<"GET">>], Req, State}.
+
+is_authorized(Req, State) ->
+    case cowboy_req:parse_header(<<"authorization">>, Req) of
+        {ok, {<<"basic">>, {Email, Password}}, Req2} ->
+            case db_user:authenticate(Email, Password) of
+                {ok, User} ->
+                    {true, Req2, #{user => User}};
+                _ -> {{false, <<"Basic realm=\"holidayping\"">>}, Req2, State}
```

Comment:
Actually my idea wasn't to use basic auth everywhere. I wanted to use token auth kind of like OAuth so you send a token in the `Authorization: Bearer ...` header in the other handlers. But this one handler is the one that generates that token (i.e. what will be called for login), so you need to pass the user/password to get it. Basic Auth seemed a better fit for the rest behavior that sending user/pass in the body.